src/parser.ts

```diff
@@ -2169,7 +2169,7 @@ export class Parser extends DiagnosticEmitter {
             tn.range()
           );
         }
-        returnType = this.parseType(tn, name.kind == NodeKind.CONSTRUCTOR || isSetter);
+        returnType = this.parseType(tn, isSetter || name.kind == NodeKind.CONSTRUCTOR);
```

Comment:
What's the improvement here?

Reply:
Oh, read it the wrong way around. Nvm

Reply:
just reorder by complexity. `isSetter` much more cheap checks in term of perf